let's release this thing !

carnaval opened the pull request:
everyone cheer, Jake is near the finish line
- [x] Rendering of `Mmap.*`
- [ ] Sphinx search result/permalink of `Base.process_messages`
- [ ] Sphinx search result for macros (e.g. `@assert` `@nloops`)
- [ ] Stdlib doc for the doc system?
- [x] `@repl` documentation for macrocalls

jakebolewski committed:
remove Explicit markup ends without a blank line; unexpected unindent warning

jakebolewski committed:
regenerate stdlib documentation, 377 -> 50 warnings

jakebolewski committed:
get rid of some markup that is causes errors in MD -> RST conversion

50 -> 34 errors

jakebolewski committed:
remove Lazy dependency in convertmd.jl, format 4 spaces

jakebolewski committed:
more unindent fixes, 34 -> 25 errors

jakebolewski committed:
remove inline emphasis warning

jakebolewski committed:
add newline after all modules

jakebolewski committed:
remove Sorting Algorithms link

tkelman commented:
This looks okay on formatting fixes but there are still lingering doc content problems not addressed yet.

ViralBShah commented:
Can I merge #12766 before this one, or should we wait?

ScottPJones commented:
Some of us must have #12766 working (to avoid GPL problems)

StefanKarpinski commented:
#12766 is a bug fix and can happen any time.

jakebolewski committed:
add line end to sign

jakebolewski committed:
make sure that a module is always followed by a newline

jakebolewski committed:
remove doc/convertmd.jl

jakebolewski committed:
fix ifft docstring

jakebolewski committed:
fix convert documentation

jakebolewski committed:
fix manage docstring

jakebolewski committed:
fix formatting of channel docstring

jakebolewski committed:
fix DateTime docstring

jakebolewski committed:
fix dupicate printing for rst method docs

carnaval committed:
add latex rst rendering, fix #12775

carnaval committed:
double backticks for rst

jakebolewski committed:
split up plain docstrings

ihnorton committed:
Correct docstring for @__FILE__ ref #12727

jakebolewski committed:
add arguemnts to Pkg methods in helpdb

carnaval committed:
Docstrings document signatures instead of methods

carnaval committed:
remove the signature from md embeded rst

carnaval committed:
half fix the signatures

jakebolewski committed:
add types to helpdb methods

jakebolewski committed:
add missing && and || syntax

carnaval commented:
So, when Tony said "lingering docs problem", I'm not sure we realized how bad it was.
This PR now addresses most of what we could find that is critical for having a decent manual. Many of the docstrings are still rst so somewhat poor REPL experience.

In particular :
- fix various missing docs
- associate function docstring to tuple types instead of methods so that you can document many methods at once with a wider tuple (the REPL help matching function is still a bit rough and could use some love)
- separate docstrings in helpdb so that each map to a single rst stdlib entry (a lot of signatures are not the best, could use people going through to look)

A lot of this was done _by hand_. I think this should make us question our release policy. We basically got nothing out of this doc transition and had to rush it.

I want my last 12 hours back. And I'm pretty sure Julia Computing owes us a bottle.
Never again...

jakebolewski commented:
Drinking bourbon for 12 hours straight is also not healthy.

timholy commented on base/docs/Docs.jl:
```diff
@@ -141,42 +144,44 @@ end
+    sort!(fd.order, lt = (a,b) -> ccall(:jl_type_morespecific, Int32, (Any,Any), a, b) > 0)
+    if haskey(ENV, "GO_AWAY") && haskey(fd.meta, sig) # temporary
+        error("Adding doc to $f($sig) forbidden GO AWAY")
+    end
```
It _does_ look like you were having fun.

carnaval committed:
make code blocks work again

jakebolewski committed:
update tests

jakebolewski committed:
fix failing doctest

jakebolewski committed:
filter generic functions to see if sig is applicable

jakebolewski committed:
display doc results in order of most -> least specific

yuyichao commented on doc/stdlib/base.rst:
```diff
@@ -23,171 +23,98 @@ Getting Around
    .. code-block:: julia
 
-       whos([Module,] [pattern::Regex])
+       whos([io,] [Module,] [pattern::Regex])
```
This should probably be an additional signature instead of a code block.

yuyichao commented on doc/stdlib/base.rst:
```diff
@@ -734,90 +418,60 @@ Types
-       sizeof(type)
 
    Size, in bytes, of the canonical binary representation of the given type, if any.
 
```
The code block below (`sizeof(s::AbstractString)`) should be another signature of `sizeof` instead of a code block.

yuyichao commented on doc/stdlib/base.rst:
```diff
@@ -2181,9 +1329,6 @@ Reflection
 .. function:: functionloc(m::Method)
```
duplicated `functionloc(m::Method)` and signature in code block above and below

yuyichao commented on doc/stdlib/collections.rst:
```diff
@@ -425,9 +302,6 @@ Iterable Collections
 .. function:: maximum(A, dims)
```
Duplicated `maximum(A, dims)` in code blocks above and below.

yuyichao commented on doc/stdlib/collections.rst:
```diff
@@ -464,9 +332,6 @@ Iterable Collections
 .. function:: minimum(A, dims)
```
Duplicated `minimum(A, dims)` in code blocks above and below

yuyichao commented on doc/stdlib/collections.rst:
```diff
@@ -740,9 +464,6 @@ Iterable Collections
 .. function:: sumabs(A, dims)
```
Duplicated `subabs(A, dims)` in code blocks above and below

yuyichao commented on doc/stdlib/collections.rst:
```diff
@@ -1257,9 +738,6 @@ Iterable Collections
 .. function:: collect(element_type, collection)
```
Duplicated `collect(element_type, collection)` in code blocks above and below

yuyichao commented on doc/stdlib/collections.rst:
```diff
@@ -1691,74 +998,50 @@ Set-Like Collections
 
    Construct the symmetric difference of elements in the passed in sets or arrays. Maintains order with arrays.
 
 .. function:: symdiff!(s, n)
```
Something is messed up with the three signatures and the code blocks for `symdiff!`

kshyatt commented on doc/stdlib/math.rst:
```diff
@@ -2421,9 +1811,6 @@ Statistics
 .. function:: middle(x, y)
```
Something has also gone wrong here

kshyatt commented on doc/stdlib/math.rst:
```diff
@@ -2502,102 +1883,60 @@ Statistics
 
    .. Docstring generated from Julia source
-   .. code-block:: julia
-
```
quantile is broken too

kshyatt commented on doc/stdlib/arrays.rst:
```diff
@@ -211,80 +150,38 @@ Constructors
 
    .. Docstring generated from Julia source
-   .. code-block:: julia
-
```
Duplicated blocks for `ones`

jakebolewski commented:
Documentation is currently broken for the following methods: 

(Please add to this list as you discover them)

- [ ] sizeof
- [ ] functionloc 
- [ ] ones
- [ ] sumabs
- [ ] sumabs!
- [ ] symdiff!
- [ ] middle

The following methods are currently missing documentation:

- [ ] \
- [ ] .\

yuyichao commented:
There's some missing code block quoting in [`Base.Cartesian`](http://julia.readthedocs.org/en/jcb-docfixes/devdocs/cartesian/)

yuyichao commented:
There's a mysterious `- _man-packages:` in the FAQ above [`Packages and Modules`](http://julia.readthedocs.org/en/jcb-docfixes/manual/faq/#packages-and-modules)

yuyichao commented:
The first "results in" for [async io](file:///home/yuyichao/projects/julia/docfixes/doc/_build/html/manual/faq.html#why-do-concurrent-writes-to-the-same-stream-result-in-inter-mixed-output) should probably have a code block.

yuyichao commented:
"Package Development" and "Memory allocation analysis" have their own link in the index page but are only sub pages. All other link in the index page have their own pages. Not sure if this inconsistency matters.

yuyichao committed:
Fix function matching regex and run genstdlib to fix code blocks in Base.Cartesian

yuyichao committed:
Fix tag in Package FAQ

yuyichao committed:
Fix code block in FAQ

yuyichao committed:
Fix code block in Dates

yuyichao committed:
Doc of Period types

nolta committed:
fix `ones`

nolta committed:
fix whos,sizeof,functionloc,sumabs docstrings

nolta committed:
fix symdiff!,middle,quantile,collect docstrings

jakebolewski committed:
fix maximum, minimum, sort docstrings

jakebolewski committed:
remove redundant scale!(b,A) doc

jakebolewski committed:
fix svd docstring

nolta committed:
fix readdlm docstrings

nolta committed:
fix latex typo in expm docstring

nolta committed:
fix schurfact,linreg,svdvals,PipeBuffer docstrings

jakebolewski committed:
fix ldltfact docstring

jakebolewski committed:
add DateTime Type docs

yuyichao commented:
(Might not be regression) Function head for `Mmap.`* looks bad. The function name and the module name appears in two boxes.

![Mmap.Anonymous](https://cloud.githubusercontent.com/assets/712232/9646278/2c788f34-51a0-11e5-9340-f602af75878f.png)

yuyichao commented:
`Base.process_messages` shows up as `Base.Base.process_messages` in search in additional to the double box issue above.

yuyichao committed:
Fix code block for *(s, t)

yuyichao committed:
Fix code block for `@sprintf`

Also let the genstdlib.jl script remove the hidden signatures in cartesian.rst since we need to fix that properly anyway and is keeping track of it somewhere else.

yuyichao committed:
Fix doc of unsafe_copy!

yuyichao committed:
Fix doc for connect

yuyichao committed:
Add missing optional parameter for whos

yuyichao committed:
Remove duplicated entry for scale and scale!, wait for the genstdlib.jl improvement to fix all missing synonyms

yuyichao committed:
Fix @__FILE__ doc and add missing doc for @__LINE__

yuyichao committed:
Remove extra code block for Period types. The doc is clear enough without them.

yuyichao committed:
Fix basedocs, sync doc for ccall

yuyichao committed:
Fix Date types

yuyichao committed:
Fix (hopefully) all synonyms

yuyichao committed:
Consistent empty lines after ".. Docstring generated from Julia source"

yuyichao commented:
All the regressions I've found (except the display/search issue with `Mmap.*` and `Base.process_messages`) should be fixed now!!

Another non-regression issue is the document of `issubset`. There're two entries for it in `collections.rst` which aren't exactly the same and one of them include "not issubset" at the same time.

yuyichao commented:
Note that I didn't compare the current rst doc to the 0.3 one, the 0.4 doc before the docsystem change or all the entries in the base doc so there might still be missing documentations.

yuyichao committed:
Fix duplicated docs for ldltfact

ViralBShah commented:
Should we merge this PR and continue the remainder of the process over the RC phase?

yuyichao committed:
Sync and fix code blocks for round

yuyichao committed:
Clean up function doc more aggressively in order to catch more missing doc in helpdb.jl

yuyichao commented:
I think this PR can technically be merged at any time since it should be strictly an improvement. However, I think merging this now won't have too much benefit either. I'm pretty sure there's many issues that are not discovered (mainly because I find new ones every time I read through the rendered html doc).

What would be helpful is if more people can help reviewing this. More specifically,

1. Go to http://julia.readthedocs.org/en/jcb-docfixes/ and read through **very carefully** the doc to see if there's anything that looks weird. (Missing code quoting, repeating docs, function head appears in code blocks)
2. Check if any document is missing docs. This could probably be done automatically in the `genstdlib.jl` script. Not sure if it is still necessary to compare to the 0.3 doc. (i.e. if we lost some doc from both the rst and the `helpdb.jl`)

yuyichao committed:
Fix keyword arguments doc for eigs

KristofferC commented:
http://julia.readthedocs.org/en/jcb-docfixes/stdlib/c/#Base.unsafe_convert
http://julia.readthedocs.org/en/jcb-docfixes/stdlib/c/#Base.cconvert

has " instead of `

bicycle1885 commented:
* http://julia.readthedocs.org/en/jcb-docfixes/stdlib/file/#Base.@__FILE__
* http://julia.readthedocs.org/en/jcb-docfixes/stdlib/file/#Base.@__LINE__

I think parentheses are not needed to call these macros.

bicycle1885 commented:
* http://julia.readthedocs.org/en/jcb-docfixes/stdlib/file/#Base.ctime
* http://julia.readthedocs.org/en/jcb-docfixes/stdlib/file/#Base.mtime
* http://julia.readthedocs.org/en/jcb-docfixes/stdlib/file/#Base.filemode
* http://julia.readthedocs.org/en/jcb-docfixes/stdlib/file/#Base.filesize

I think `stat(file).foo` should be surrounded by backquotes.

bicycle1885 commented:
[] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/file/#Base.mv

Use `dst` instead of `dest`, and surround them with backquotes.

yuyichao committed:
Fix code quote for cconvert and unsafe_convert

yuyichao commented:
@bicycle1885 The syntax for todo list is `- [ ]` note the space between `-` and `[` as well as `[` and `]`

bicycle1885 commented:
@yuyichao Thank you! Fixed.

bicycle1885 commented:
- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/io-network/#Base.@printf

Quote `args`.

yuyichao committed:
Fix quoting and escapes for package functions

KristofferC commented:
I think you could just update your previous comment to make it more compact.

yuyichao committed:
Fix quoting for @printf

bicycle1885 commented:
OK, I'll do that from now on.

- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/io-network/#Base.setopt

Quote default values: `true` and `false`.

yuyichao committed:
Add many missing quotes

yuyichao committed:
More missing quotes

yuyichao committed:
Fix capitalization of the Function type

yuyichao committed:
Fix quote for rem

bicycle1885 commented:
- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/parallel/#Base.@parallel

`@parallel` should be surrounded by backquotes.

---

- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/linalg/#Base.ordschur

An inline code is broken (scroll down to `(GS[:S], GS[:T])`).

![screen shot 2015-09-03 at 23 46 23](https://cloud.githubusercontent.com/assets/905683/9661784/400ac4d4-5297-11e5-95e4-11de3ebe8d6d.png)

yuyichao committed:
Fix quote for @parallel and type parameter for UTInstant

yuyichao committed:
Fix code quote for ordschur

bicycle1885 commented:
Asynchronously updating a TODO list is not safe :cry: 

- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/strings/#Base.isgraph

`isgraph(c)==true` should be surrounded by backquotes.

---

- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/strings/#Base.utf16

`utf16(s)` should be surrounded by backquotes.
And unpaired backquotes around `UTF16String(A)`?

![screen shot 2015-09-04 at 0 07 09](https://cloud.githubusercontent.com/assets/905683/9662098/ef290bf0-5298-11e5-8a5f-79db8aa2c1dc.png)

yuyichao committed:
Remove parenthesis for @__LINE__ and @__FILE__

catawbasam commented:
In http://julia.readthedocs.org/en/jcb-docfixes/stdlib/strings/

- [ ] match()  -- RegexMatch shoud be quoted

- [ ] chop(), chomp(), ind2chr(), chr2ind(), isvalid()  -- add a period at the end of the description.

- [ ] isgraph(c::Union{Char, AbstractString})  -- isgraph(c)==true should be backquoted.

- [ ] isvalid(str, i) -- should be moved adjacent to other isvalid() methods.

- [ ] utf16(s) --  UTF16String(A)`  missing initial backquote.

(these mostly look like pre-existing issues rather than conversion bugs)

bicycle1885 commented:
- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/numbers/#Base.hex2bytes
- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/numbers/#Base.bytes2hex

`Array{UInt8, 1}` and `ASCIIString` should be surrounded by backquotes, respectively.

---

- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/numbers/#Base.Inf

Type names (`Float(16|32|64)`) should be surrounded by backquotes for each constant (`Inf`, `Inf32`, `Inf16`, `NaN`, `NaN32` and `NaN16`).

---

- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/numbers/#bigfloats

Four `BigFloat`s in this section should be surrounded by backquotes.

yuyichao committed:
Fix more code quote

catawbasam commented:
http://julia.readthedocs.org/en/jcb-docfixes/stdlib/base/
### Getting Around

- [ ] `@edit()` -- missing period at end
- [ ] `@less()`-- missing period at end
- [ ] require(module::Symbol)  -- typo "(for exmple)"
- [ ] `@show()` -- missing period at end

catawbasam commented:
@yuyichao are you just looking for list of problems, or can we also be correcting them using Edit on Github?

yuyichao commented:
Feel free to submit PR against this branch. (not the master branch)

yuyichao commented:
And for most of the doc fixes. Make sure to edit them in the code and sync it to the doc.

Rough procedure.

1. Locate the problem in the rst file.
2. If it's in the doc of a function with `.. Docstring generated from Julia source`. Locate them in the julia file (usually `base/doc/helpdb.jl`, and this is the reason we don't move them inline yet). If it is not generated from julia file, edit the rst file directly and go to 6.
3. Update the doc in the julia file
4. Compile julia (this takes a long time and is why it takes me a while to commit each fix)
5. Run `./julia doc/genstdlib.jl`.
6. Check if the rst doc is fixed.
7. Commit and push out.

yuyichao commented:
Optionally, you can also run `make -C doc/` and checkout the generated html doc in `doc/_build`

yuyichao committed:
Consistent quoting and fix some math intervals

carnaval commented:
note that if you don't want to bootstrap every time you can work around that by removing helpdb from sysimg.jl and adding `eval(Base, :(include("helpdb.jl"))` in the genstd script IIRC

yuyichao commented:
True. That should work and can probably be much faster as long as it's not inlined doc string.

Anyway, I'm going to do something else this afternoon (next few hours). Feel free to fix whatever comes up and don't worry about conflicting with what I'm fixing.

yuyichao commented:
The genstdlib is not doing anything too fancy so small fixes (missing period, typo, missing quote) could be done through the web interface as long as both the rst and the julia doc are updated.

jakebolewski commented:
it is probably easier to just report problems here, its easy to fix them once they have been identified

jakebolewski commented:
Another very important thing we need help testing is `help?>` behavior.  The way this works now is very different than the previous behavior in that it tries to return the most specific help for a given input (first) but all applicable help for that method as well given the current signature.

For example:
```julia
help?> fft
```
in addition to looking at the types of a methods arguments:

```julia
julia> a = Dict(1=>2)
Dict{Int64,Int64} with 1 entry:
  1 => 2

help?> a[1]
  ..  getindex(collection, key...)

  Retrieve the value(s) stored at the given key or index within a collection.
  The syntax ``a[i,j,...]`` is converted by the compiler to
  ``getindex(a, i, j, ...)``.

julia> b = Int[1,2,3]
3-element Array{Int64,1}:
 1
 2
 3

help?> b[1]
  ..  getindex(A, inds...)

  Returns a subset of array ``A`` as specified by ``inds``, where each ``ind`` may be an ``Int``, a ``Range``, or a ``Vector``. See the manual section on :ref:`array indexing <man-array-indexing>` for details.

  ..  getindex(collection, key...)

  Retrieve the value(s) stored at the given key or index within a collection.
  The syntax ``a[i,j,...]`` is converted by the compiler to
  ``getindex(a, i, j, ...)``.

help?> Int[1,2,2]
  ..  getindex(type[, elements...])

  Construct a 1-d array of the specified type. This is usually called with the syntax ``Type[]``. Element values can be specified using ``Type[a,b,c,...]``.

  ..  getindex(collection, key...)

  Retrieve the value(s) stored at the given key or index within a collection.
  The syntax ``a[i,j,...]`` is converted by the compiler to
  ``getindex(a, i, j, ...)``.
```

tkelman commented:
Something that I really miss from the old repl help is the summary display of fields for types where the constructor doesn't have any more specific docstring. For example, on 0.3:
```
help?> SparseMatrixCSC
INFO: Loading help data...
DataType   : SparseMatrixCSC{Tv,Ti<:Integer} (constructor with 1 method)
  supertype: AbstractSparseArray{Tv,Ti<:Integer,2}
  fields   : (:m,:n,:colptr,:rowval,:nzval)
```

Would have to check if this branch does anything better than master for that, but I'd consider this a regression from the conversion.

jakebolewski commented:
No, it gives no output.  The this was the old behavior as well (before this PR).  I also removed the method errors you would sometimes get when trying query help for a method that doesn't exist.

jakebolewski commented:
@tkelman that should be easy to add back.

tkelman commented:
@MichaelHatherly agreed in https://github.com/JuliaLang/julia/issues/12437#issuecomment-128681258, doesn't technically have to be done in this PR but you brought up repl help and I hadn't opened a separate issue on that one yet.

jakebolewski committed:
some minor fixes in string stdlib section

my-little-repository commented:
is there any interest for a list of missing permalinks? Not sure how/where it can be fixed.
All missing permalinks in http://julia.readthedocs.org/en/jcb-docfixes/stdlib/arrays/

ind2sub(a, index) → subscripts
zeros(A)
ones(A)
eye(m, n)
getindex(A, inds...)
find(f, A)
findfirst(A, v)
findfirst(predicate, A)
findlast(A, v)
findlast(predicate, A)
findnext(predicate, A, i)
findnext(A, v, i)
findprev(predicate, A, i)
findprev(A, v, i)
rot180(A, k)
rotl90(A,k)
rotr90(A,k)
nthperm(p)
partitions(n,m)
partitions(array)
partitions(array, m)
rol!(B::BitArray{1}, i::Integer) → BitArray{1}
ror!(B::BitArray{1}, i::Integer) → BitArray{1}
sparsevec(D::Dict[, m])
sparse(A)
sparsevec(A)

catawbasam commented:
http://julia.readthedocs.org/en/jcb-docfixes/stdlib/base/#types
### Types

<b>Note:</b> Entries in this section mix the use of 'T' and 'type'.  Maybe just use 'T'?

- [ ] super(T::DataType) -- DataType and T should be quoted.
- [ ] subtypes(T::DataType) -- DataType and T should be quoted.
- [ ] typemin(type)  -- type should be quoted
- [ ] typemax(type)  -- type should be quoted
- [ ] realmin(type)  -- type should be quoted;  missing period
- [ ] realmax(type)  -- type should be quoted;  missing period
- [ ] maxintfloat(type)  -- type should be quoted;  missing period
- [ ] sizeof(type)  -- type should be quoted
- [ ] getfield(value, name::Symbol) -- value should be quoted
- [ ] fieldtype(type, name::Symbol | index::Int)  -- name, index and type should be quoted

ssfrr commented:
On this branch I get an error when looking up help for a method in a package. It works if you prefix the package name with `Main.` (idea from @carnaval and @jakebolewski)

```julia
julia> using IterativeSolvers

help?> IterativeSolvers.randnn
ERROR: UndefVarError: IterativeSolvers not defined

help?> Main.IterativeSolvers.randnn
  Randomized Gaussian matrices normalized by column

  Input:     el: element type     m: number of rows     n: number of columns or nothing     normalize: whether or not to normalize (default: true)

  Output:     Ω: a matrix of dimensions m x n containing Gaussian random numbers of     type el.

julia> 
```

jakebolewski commented:
The rendering (or parsing) of the docstrings seems to be stripping out the original newlines.

catawbasam commented:
##### Standard Library Essentials, final sections:

http://julia.readthedocs.org/en/jcb-docfixes/stdlib/base/#syntax
- [ ] eval([m::Module, ]expr::Expr) -- capitalize and quote module
- [ ] esc(e::ANY) -- quote Expr
- [ ] parse(str, start; greedy=true, raise=true)-- lowercase and quote start
- [ ] parse(str; raise=true) --  Substitute 'expression string' for 'whole string' to match verbiage in the other parse() entry.
- [ ] setenv(command, env; dir=working_dir) -- quote command
- [ ] `@timev()` -- End "This is a verbose version of the `@time` macro, it"... should be "This is a verbose version of the `@time` macro. It"...

http://julia.readthedocs.org/en/jcb-docfixes/stdlib/base/#errors
- [ ] Note:  The treatment of message parameters is inconsistent.  We have msg, message, text as labels.
Some error functions are documented as taking msg, others [msg], and still others no message parameter.

- [ ] `@assert` cond [text] -- Define the [text] parameter.  
- [ ] ArgumentError(msg) -- Define the msg parameter.  (there are several of these cases)
- [ ] SystemError(prefix::AbstractString[, errnum::Int32])  -- 'errnum' and 'errno' consistency?
- [ ] TypeError(func::Symbol, context::AbstractString, expected::Type, got) -- 'got' looks like it should be 'got::Type'

http://julia.readthedocs.org/en/jcb-docfixes/stdlib/base/#events
- [ ] Timer(callback::Function, delay, repeat=0) -- quote callback and delay
- [ ] Timer(delay, repeat=0) -- define time units of delay

http://julia.readthedocs.org/en/jcb-docfixes/stdlib/base/#reflection
- [ ] Multiple cases where Module, Symbol and Function should be capitalized and quoted.
- [ ] fieldnames(x::DataType) -- 'data type' should be DataType and quoted

http://julia.readthedocs.org/en/jcb-docfixes/stdlib/base/#internals
- [ ] Multiple cases of missing period at end of definition.

mdcfrancis commented:
There are some datatype errors in the Associative docs - I fixed them in the following https://github.com/JuliaLang/julia/pull/12931 - this request includes an additional export so it might be worth folding the doc changes in here ?

yuyichao committed:
Fix and sync doc for || and &&

yuyichao committed:
Warn about missing document

yuyichao commented:
The following methods seems to be missing some documents. The list is generated with script from the documents in the code that are not used by the rst doc.

- [ ] `reverse`
- [ ] `unsafe_checkstring`
- [ ] `skipwhitespace`
- [ ] `startswith`
- [ ] `checkstring`
- [ ] `encode_to_utf8`
- [ ] `skipblank`
- [ ] `$`
- [ ] `doc!`
- [ ] `Base.@doc`
- [ ] `readuntil`
- [ ] `Base.Docs.@html_str`
- [ ] `unindent`
- [ ] `withstream`
- [ ] `Base.Docs.@text_str`
- [ ] `Base.@r_str`
- [ ] `eatindent`
- [ ] `catdoc`
- [ ] `doc`
- [ ] `indentation`
- [ ] `get!`
- [ ] `checkbounds`
- [ ] `linecontains`
- [ ] `InitError`
- [ ] `Base.Docs`
- [ ] `parse_inline_wrapper`
- [ ] `encode_to_utf16`
- [ ] `summarysize`

yuyichao committed:
Fix some missing doc strings

yuyichao committed:
Add doc for summarysize

yuyichao committed:
Use quote and add missing doc for InitError

JeffreySarnoff commented:
in commit 092bcf4 (3 days old)
getting help on Pkg no longer responds with Pkg.add ..
help?> pkg suggests Pkg
help?> Pkg  just returns search: Pkg

yuyichao committed:
Add document for string macro @r_str

yuyichao committed:
Use @r_str due to rendering and searching issue...

jakebolewski committed:
add some missing quotes

jakebolewski committed:
some getting started section fixes

jakebolewski committed:
some types section fixes

jakebolewski committed:
stdlib base syntax fixes

jakebolewski committed:
more stdlib essentials fixes and improvements

jakebolewski committed:
report exported / unexported missing module documentation

jakebolewski commented:
Ok, thanks everyone who took the time to report regressions and other documentation errors.  Hopefully all the ones  that have been reported have been addressed.  I still have to fix the issue reported by @ssfrr, but after that this should be largely "done".

jakebolewski committed:
properly escape input in  for toplevel symbols / fields

remove debugging code

jakebolewski commented:
Ok, I think this is pretty much done now.  Needs another round of review.

tkelman commented:
What happened to the commit that had the message "NEVER AGAIN"? Was that completely superseded by later regenerations?

jakebolewski commented:
it was superseded

carnaval commented:
noooooo my line count :(

jakebolewski commented:
All your lines are there, most arbitrary metrics are preserved.  I grifted you have ~1 commit in a rebase.

MichaelHatherly commented:
Minor formatting issue in http://julia.readthedocs.org/en/jcb-docfixes/stdlib/parallel/#Base.fetch where the first item in the list isn't rendered properly. Other than that looks great, thanks @jakebolewski, @carnaval, @yuyichao, @nolta, @ihnorton, and everyone else.

yuyichao commented:
`@assert` and possibly others doesn't show up in search result

yuyichao commented:
`watch_file` has wrong argument name

yuyichao commented:
`popdisplay` is broken

yuyichao commented:
Argument description for `fetch` has weird formatting

bicycle1885 commented:
More problems I've found in the current document. I'm sorry if these have been fixed.

- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/parallel/#Base.current_task
- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/parallel/#Base.@schedule
- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/parallel/#Base.@task

Quote `Task`.

---

- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/parallel/#Base.Channel{T}

Quote `Channel`.

---

- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/parallel/#Base.@everywhere

Quote `CompositeException`.

---

- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/parallel/#Base.SharedArray

Quote `SharedArray`.

---

- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/parallel/#Base.close

Dup of "on a".

---

- [ ] http://julia.readthedocs.org/en/jcb-docfixes/stdlib/parallel/#Base.Base.process_messages

Quote `AsyncStream`.

BTW, is there any reason not to export `AsyncStream` from `Base`?  A recent master branch (877b088) does not export it.

yuyichao committed:
Fix doc for popdisplay and watch_file

yuyichao committed:
Fix argument doc for fetch

yuyichao committed:
Fix quoting for Channel and Task

yuyichao committed:
Fix more quoting in parallel

yuyichao commented:
I did a comparasion with the 0.3 doc and 0.4 doc before the switch and I think we are not missing anything anymore.

yuyichao committed:
Fix doc for summarysize

yuyichao committed:
Consistent arguments names

yuyichao committed:
Fix a few quoting

yuyichao committed:
Missing quote for `A`

yuyichao committed:
More quotes

yuyichao committed:
Still about quotes in document

yuyichao committed:
Update doc for cd()

yuyichao committed:
Quote in Dates

jakebolewski committed:
fix macro repl help lookup, add test

quinnj commented:
![let's celebrate y'all](https://media.giphy.com/media/xeXEpUVvAxCV2/giphy.gif)

nolta commented:
:cocktail: :cocktail: :cocktail:

timholy commented:
Our heroes!

bicycle1885 commented:
:beers: :sake: :wine_glass:

ScottPJones commented:
Cheers to all the great work done here by some pretty dedicated people! :clap::clap::clap:

Sisyphuss commented:
Cheers!

KristofferC commented:
![fig_022](https://img.4plebs.org/boards/pol/image/1411/67/1411676201464.gif)

ScottPJones commented:
OK, was this the last thing holding up branching a 0.4 pre-release?
(please don't call it RC, unless it really is a candidate [with no changes] for the first release)
If so, I'm going to get sloppily drunk celebrating!

jakebolewski commented:
Check your milestones before you take to the bottle.

tkelman commented:
I'll open a separate issue for the type summary in repl help.

ScottPJones commented:
@jakebolewski https://github.com/JuliaLang/julia/issues/11536#issuecomment-135592067 the roadmap for 0.4 was closed, and no other was reopened, so, how do I check the milestones???

quinnj commented:
https://github.com/JuliaLang/julia/issues?q=is%3Aopen+is%3Aissue+milestone%3A0.4.0

johnmyleswhite commented:
https://github.com/JuliaLang/julia/milestones/0.4.0
https://github.com/JuliaLang/julia/milestones/0.4.x

ScottPJones commented:
Thanks! I hadn't seen that feature of GitHub before, had just been following the 0.4 roadmap issue, but now I'm going to start drinking, after seeing how many things still to be done, most dating from 1-3 years ago!

ViralBShah commented:
:clap:

StephenVavasis commented:
I hope I am posting this in the right place: if you are still looking for documentation issues, including the library documentation, then I would like to bring up an issue related to:

https://github.com/JuliaLang/DataStructures.jl/pull/114

and the work of @mbauman. The documentation states that an IntSet holds 'nonnegative' integers, and yet trying to insert 0 into an IntSet causes a deprecation warning in all recent masters.

My own preference is to remove the deprecation warning (see the discussion mentioned above), but if the deprecation warning is not removed, then the documentation should be fixed.
